Backend/scala-backend/app/controllers/SurveyController.scala

```diff
@@ -187,6 +187,7 @@ class SurveyController @Inject() (
   def startPartitionAlgorithm(classId: Int): Unit = {
     val studentsOfClass: Future[Seq[Int]] =
       studentModel.getAllSelfReportedStudentIDs(classId)
```

Comment:
Dieses future muss doch notwendigerweise beendet sein, bevor der kram unten mit dem updateGroupBelonging aufgerufen wird. Hier werden also offensichtlich noch studenten in der datenbank gefunden. Warum dann später nicht mehr??